ls1intum/Artemis
# `Development`: Refine path comparison

### Checklist
#### General


#### Server

### Motivation and Context
- Some file paths may address areas outside of the repository (invalid file paths).
- Naming file path parameters "filename" may spread the expectation, that this is just a file name and not a file path (with less things required to be considered).

### Description
The file paths now compared in an unescaped, normalised and absolute manner against the repository path in order to check whether the given path is valid. Also filename parameters were renamed to filePath in order to not spread incorrect expectations. 

### Steps for Testing
We just want to make sure that I didn't accidentally break the implementation.

Prerequisites:
- 1 Instructor
- 1 Student

After each step, commit and see if the changes were persisted by reloading
Programming Exercise (Instructor, Student):
1. Create a programming exercise with the online editor activated.
2. Participate in the programming exercise/alternatively edit the solution/template in the online editor
3. Create a folder
4. Create a folder within the folder
5. Create a file within the second folder
6. Rename the file
7. Delete the file and folders
8. No issues should occur

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| src/main/java/de/tum/in/www1/artemis/service/RepositoryService.java | 91.5% | ✅                           |
